README.md

```diff
@@ -1,58 +1,148 @@
+
+- EIDF Cloud Instance: The EIDF cloud provides robust and scalable infrastructure for data storage and computation, which is crucial for handling the large volumes of data generated by autonomous vehicles. By utilizing the EIDF cloud, the script ensures that data is securely stored and readily available for further analysis and processing.
+
+- Data Collection: The rosbags collected contain crucial data from various sensors on the vehicle, including LiDAR, cameras, and GPS. This data is essential for research and development in autonomous driving, enabling tasks such as sensor fusion, environment mapping, and behavior prediction.
+
+- Automation and Efficiency: Manually uploading large datasets can be time-consuming and prone to errors. This script automates the entire process, from compressing the rosbags to transferring them securely to the remote host machine. It also includes functionality to manage disk space efficiently on the remote machine.
+
+### 1. Dependencies
+
+
+  ##### 1. Remote host machine
+- iperf3: A tool for measuring bandwidth.
+```bash
+sudo apt-get install iperf3
+```
+- set SSH keys authentication: This should be done before using this script.
+   - create an ssh-key:
+   ```bash
+   ssh-keygen -t rsa -b 2048
+   ```
+   - share a SSH Key
+   ```bash
+   ssh-copy-id username@hostname_ip
+   ```
+
+#### 2. Remote machine
+- iperf3: A tool for measuring bandwidth.
+```bash
+sudo apt-get install iperf3
+```
+- MCAP cli: A tool for compressing rosbags
+```bash
+wget -O $HOME/mcap https://github.com/foxglove/mcap/releases/download/releases%2Fmcap-cli%2Fv0.0.47/mcap-linux-amd64 && chmod +x $HOME/mcap
+```
+### 2. Usage
+To run the script, use the following command:
+  ```bash
+  python3 upload_vehicle_data.py -config <path_to_yaml_config>
+  ```
+  or
+  ```bash
+  python3 upload_vehicle_data.py -c <path_to_yaml_config>
+  ```
+Replace <path_to_yaml_config> with the path to your YAML configuration file that contains the necessary parameters for the script.
+
+### 3. YAML Parameters
+- `remote_user` (str): Username for the remote machine.
+- `remote_ip` (str): IP address of the remote machine (vehicle PC) (default: 129.215.117.104).
+- `remote_directory` (str): Directory on the remote machine containing rosbags (default: /mnt/mydrive/rosbags).
+- `cloud_upload_directory` (str): Remote host directory for uploading compressed files (default: /mnt/vdb/data).
+- `clean_up ` (bool): Whether or not to delete all rosbags from the vehicle machine after uploading
+- `upload_attempts` (int): The number of attempts the script should make to upload each rosbag file to the cloud host. If not specified, the default value is `3`.
+- `mcap_path` (str): The binary path for the mcap cli. Can be found using `which mcap` if mcap is correctly installed.
+- `parallel_processes`: The number of parallel processes to use for compression and upload.
+YAML file example:
+```bash
+remote_user: "user namee"

```

Comment:
`"user name"`

Could you pass all of the script via a spell checker just in case we missed something?

Reply:
@ShikaiGeng Has this been adressed?

Reply:
Changed